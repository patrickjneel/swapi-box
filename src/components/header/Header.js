import React from 'react';
import './header.css';
import PropTypes from 'prop-types';


const Header = ({upDateData, favorites}) => {
  // let classChange = upDateData ? 'unselected' : 'selected'

  return (
    <div className="header">
      <button onClick={() => upDateData('people')}>
       People
      </button>
      <button onClick={() => upDateData('planets')}>
       Planets
      </button>
      <button onClick={() => upDateData('vehicles')}>
       Vehicles
      </button>
      <button onClick={() => upDateData('favorites')}>
       Favorites {favorites}
      </button>
    </div>
  );
};

export default Header;

Header.propTypes = {
  upDateData: PropTypes.func,
  favorites: PropTypes.number
};
