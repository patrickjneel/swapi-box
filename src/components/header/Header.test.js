import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';

describe('Header Test', () => {

  it('should be defined', () => {
    const renderedHeader = shallow(<Header />);

    expect(renderedHeader).toBeDefined();
  });

  it('should receive props', () => {
    const mockFunc = jest.fn();
    const renderedHeader = mount(<Header upDateData={mockFunc} />);
    
    expect(renderedHeader.props().upDateData).toEqual(mockFunc);
  });

  it('should render four buttons', () => {
    const mockFunc = jest.fn();
    const mockLocation = 'people';
    const mockFavorites = 0;
    const renderedHeader = shallow(
      <Header 
        upDateData={mockFunc} 
        favorites={mockFavorites} 
        location={mockLocation}/>
    );
    expect(renderedHeader.find('button').length).toEqual(4);

  });

  it('should match sanp shot', () => {
    const mockFunc = jest.fn();
    const mockLocation = 'people';
    const mockFavorites = 0;
    const renderedHeader = shallow(
      <Header 
        upDateData={mockFunc} 
        favorites={mockFavorites} 
        location={mockLocation}/>
    );
    expect(renderedHeader).toMatchSnapshot();
  });

});


