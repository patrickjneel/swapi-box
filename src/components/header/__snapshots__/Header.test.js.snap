// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should match sanp shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    upDateData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          People
</button>,
        <button
          onClick={[Function]}
>
          Planets
</button>,
        <button
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          onClick={[Function]}
>
          Favorites 
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites ",
            undefined,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites ",
          undefined,
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            People
</button>,
          <button
            onClick={[Function]}
>
            Planets
</button>,
          <button
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            onClick={[Function]}
>
            Favorites 
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites ",
              undefined,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            undefined,
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
